src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -49,32 +51,35 @@ public class SecurityManager {
             return new ClaimsPrincipal(identity);
         }
 
-        private uint CreateProfile(string username, string domain, string password,out StringBuilder profilePath, out bool profileExists) {
-            uint error = 0;
-            profilePath = new StringBuilder();
-            profileExists = false;
-            using (NamedPipeClientStream client = new NamedPipeClientStream("RUserCreatorPipe"))
-            using (BinaryReader reader = new BinaryReader(client))
-            using (BinaryWriter writer = new BinaryWriter(client)) {
+        private async Task<JArray> CreateProfileAsync(string username, string domain, string password, CancellationToken ct) {
+            using (NamedPipeClientStream client = new NamedPipeClientStream("Microsoft.R.Host.UserProfile.Creator{b101cc2d-156e-472e-8d98-b9d999a93c7a}")) { 
                 try {
-                    client.Connect();
+                    await client.ConnectAsync(ct);
 
-                    writer.Write(username);
-                    writer.Write(domain);
-                    writer.Write(password);
+                    JArray dataArray = new JArray();
```

Comment:
Might be easier and more convenient to just shove all pieces in a struct and serialize that as a single object; and same thing for the response. Basically, the same way we do e.g. `SessionCreateRequest`.